org.eclipse.lsp4e/src/org/eclipse/lsp4e/LSPEclipseUtils.java

```diff
@@ -862,16 +871,19 @@ private static String getFileName(@NonNull IDocument document) {
 	public static List<IContentType> getDocumentContentTypes(@NonNull IDocument document) {
 		List<IContentType> contentTypes = new ArrayList<>();
 
-		ITextFileBuffer buffer = FileBuffers.getTextFileBufferManager().getTextFileBuffer(document);
-		if (buffer != null) {
-			try {
-				// may be a more specific content-type, relying on some content-type factory and actual content (not just name)
-				IContentType contentType = buffer.getContentType();
-				if (contentType != null) {
-					contentTypes.add(contentType);
+		ITextFileBufferManager bufferManager = FileBuffers.getTextFileBufferManager();
+		if (bufferManager != null) {
+			ITextFileBuffer buffer = bufferManager.getTextFileBuffer(document);
+			if (buffer != null) {
+				try {
+					// may be a more specific content-type, relying on some content-type factory and actual content (not just name)
+					IContentType contentType = buffer.getContentType();
```

Comment:
seeing error messages here caught line 884 and thrown from line 885 in Test log.

_org.eclipse.core.internal.resources.ResourceException <file> )[368]: java.lang.Exception: Resource '<file>' does not exist.
	at org.eclipse.core.internal.resources.ResourceException.provideStackTrace(ResourceException.java:42)
	at org.eclipse.core.internal.resources.ResourceException.<init>(ResourceException.java:38)
	at org.eclipse.core.internal.resources.Resource.checkExists(Resource.java:330)
	at org.eclipse.core.internal.resources.Resource.checkAccessible(Resource.java:204)
	at org.eclipse.core.internal.resources.File.getContentDescription(File.java:253)
	at org.eclipse.core.internal.filebuffers.ResourceTextFileBuffer.getContentType(ResourceTextFileBuffer.java:201)
	at **org.eclipse.lsp4e.LSPEclipseUtils.getDocumentContentTypes(LSPEclipseUtils.java:880)**_